.github/workflows/non-release.yaml

```
@@ -0,0 +1,24 @@
+name: Build
+on:
+  push:
+  pull_request:
```

```suggestion
     branches:
        - master
        - 'release-v*'
```

Similar to https://github.com/gardener/gardener-extension-registry-cache/pull/420#discussion_r2266152607